### ISSUE TITLE
Custom increment/decrement durations for custom view

### ISSUE THREAD

**Issue by highway11** [fullcalendar/fullcalendar-scheduler#36]
How would I make the previous and next buttons move only half the amount of time? For example I have a "2 month" view but would like the next button to only move the view forward 1 month. 

So if it is currently displaying Jan-Feb I want it to move one month to Feb-March instead of March-April.

 This allows the user to be able to easily drag items between months. Right now they will always have a hard time dragging an event from one month to the next

**Comment by arshaw** [comment]
i understand the need for this. this might be considered more of a "core" feature (not just scheduler)

**Comment by CodyMorris** [comment]
Hey @arshaw , as always, love the plugin.

I'm looking to implement something similar, are you waiting for acceptance as a "core" feature?  Would love to hear how this is proceeding

**Comment by supairish** [comment]
@CodyMorris Did you find a workaround for this?  I'm in need of basically the same thing.

**Comment by kwiliarty** [comment]
I have a workaround that seems to be doing the job for me. My case is slightly different but similar enough. I want to display an academic year calendar that starts in September and ends in August. The snippets below are all from the settings passed to fullcalendar. I did not have to modify any core files. First, I create a two-year custom view:
```
views: {
  academicYear: {
    type: 'list',
    duration: {year: 2},
  },
},

```
Next I filter out the events before September of the first year and after August of the second:
```
eventRender: function(event, element, view) {
  if (view.type == 'academicYear') {
    var firstDay = moment().year(view.start.year()).month('September').date(1).startOf('day');
    var lastDay = firstDay.clone().add(1, 'years').subtract(1, 'days');
    if (event.start.isBefore(firstDay, 'day') || event.start.isAfter(lastDay, 'day')) {
      return false;
    }
  }
}
```
And finally I set a new click behavior for the "prev" and "next" buttons:
```
viewRender: function(view, element) {
  if (view.type == 'academicYear') {
    var viewStart = moment(view.start).format();
    $('#calendar .fc-prev-button').click(function() {
      $('#calendar').fullCalendar('gotoDate', moment(viewStart).subtract(1, 'year'));
      $('#calendar').fullCalendar('render');
    });
    $('#calendar .fc-next-button').click(function() {
      $('#calendar').fullCalendar('gotoDate', moment(viewStart).add(1, 'year'));
      $('#calendar').fullCalendar('render');
    });
  }
},
```
It wouldn't hurt to have an easier way, but I'll go with this for now.

**Comment by kwiliarty** [comment]
Following up on my previous post, I believe now that I have found a workaround that does not suffer from the same performance/memory issues. I have removed the callback for `viewRender` and instead added some `customButtons`:

```
customButtons: {
  prevAcademicYear: {
    icon: 'left-single-arrow',
    click: function() {
      $('#calendar').fullCalendar('incrementDate', {year: -4});
      $('#calendar').fullCalendar('incrementDate', {year: 3});
    },
  },
  nextAcademicYear: {
    icon: 'right-single-arrow',
    click: function() {
      $('#calendar').fullCalendar('incrementDate', {year: 4});
      $('#calendar').fullCalendar('incrementDate', {year: -3});
    },
  },
},

```
It's hacky, but I find that by incrementing first beyond any displayed events and then partway back the calendar will be forced to rerender. It performs well, and the performance does not degrade after navigating forward and back several times.

**Comment by arshaw** [comment]
Released in core **v3.3.0** / Scheduler **1.6.0**
Blog post: https://fullcalendar.io/blog/2017/03/fullcalendar-33-and-scheduler-16/

[dateIncrement](https://fullcalendar.io/docs/current_date/dateIncrement/)
